routers/repo/download.go

```diff
@@ -31,6 +31,7 @@ func ServeData(ctx *context.Context, name string, reader io.Reader) error {
 	}
 
 	ctx.Resp.Header().Set("Cache-Control", "public,max-age=86400")
+	ctx.Resp.Header().Set("Content-Length", fmt.Sprintf("%d", size))
```

Comment:
if size > 0

Reply:
The content-length can be 0 if the body is empty. Maybe we can add size >= 0 if it is necessary ?

Reply:
Is it even possible that value could be less then zero?

Reply:
Clearly size < 0 is a bug with the calling code.
I think we should just log it and move on. This below is clearly the simplest solution:

```suggestion
		if size < 0 {
		ctx.Resp.Header().Set("Content-Length", fmt.Sprintf("%d", size))
	} else {
		log.Error("ServeData called to serve data: %s with size < 0: %d", name, size)
	}
```

if we ever just want to pass an undetermined size we can drop the log.Error to log.Debug.